### PR TITLE
fix: remove mui/material-ui ArrayFieldTemplate grid wrapper

### LINKED ISSUE #3863: MUI/material-ui theme's `ArrayFieldTemplate` renders items in double grid container
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

mui

### Version

5.x

### Current Behavior

For the MUI and material-ui themes, the `ArrayFieldTemplate` renders the `ArrayFieldItemTemplate` components inside of a grid container and those components also render a grid container. For custom themes that add default spacing to grid items, this can result in double negative spacing margins on the container which causes the field description text to be obscured. 

The current grid structure in ArrayFieldTemplate:

```
ArrayFieldTemplate > grid-containter > grid-container(ArrayFieldItemTemplate)
```

Changing the `ArrayFieldItemTemplate` wrapper to a grid item fixes the issue:

```
ArrayFieldTemplate > grid-containter > grid-item(ArrayFieldItemTemplate)
```

### Expected Behavior

The grid in `ArrayFieldTemplate` should be correctly rendered with the structure `ArrayFieldTemplate > grid-containter > grid-item(ArrayFieldItemTemplate)` so it is not possible for the `ArrayFieldItemTemplate` to create double negative margins and obscure the field description text.

### Steps To Reproduce

1. Create a form with a schema that renders the `ArrayFieldTemplate` in a MUI or material-ui theme e.g.

```jsx
        <Form
          schema={{
            "type": "array",
            "description": "Users who are authorized",
            "title": "Authorized Users",
            "items": [
              {
                "type": "string",
                "default": "user"
              }
            ],
            "additionalItems": {
              "title": "Additional Users",
              "type": "string"
            }
          }}
          validator={validator}
        />
```

2. Inspect the array field items wrapping container and notice the double grid container structure

grid-bug

### Environment

```markdown
- OS: Mac 13.5.1
- Node: 18.6.0
- npm: 9.5.1
```

### Anything else?

CodeSandbox: https://codesandbox.io/p/sandbox/festive-benji-jt6tjv?file=/pages/index.tsx:17,34

How a fix would look rendered:
grid-fix

### PR DESCRIPTION
**Pull request by jroebu14**
### Reasons for making this change

fixes #3863

Removes a wrapping Grid component in `ArrayFieldTemplate` that wraps `ArrayFieldItemTemplate`. This is for the [MUI](https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/mui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx#L41) and [material-ui](https://github.com/rjsf-team/react-jsonschema-form/blob/main/packages/material-ui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx#L41) themes only.

### Checklist

- [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://rjsf-team.github.io/react-jsonschema-form/docs/contributing) of the Markdown text I've added
- [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://rjsf-team.github.io/react-jsonschema-form/docs) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/main/CHANGELOG.md) with a description of the PR
- [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

File: packages/material-ui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx [hunk 1]
```diff
@@ -38,55 +38,57 @@ export default function ArrayFieldItemTemplate<
...
-          {hasRemove && (
-            <RemoveButton
-              style={btnStyle}
-              disabled={disabled || readonly}
-              onClick={onDropIndexClick(index)}
-              uiSchema={uiSchema}
-              registry={registry}
-            />
-          )}
+    <Grid item={true} xs={12}>
```
Comment (heath-freenome) [inline packages/material-ui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx:41]: What about entirely removing the Grid `container` on [line 63](https://github.com/rjsf-team/react-jsonschema-form/blob/6ffc2ceb1f24e2661d0acbc0c3e09e5e443e9f27/packages/material-ui/src/ArrayFieldTemplate/ArrayFieldTemplate.tsx#L63) of `ArrayFieldTemplate` instead?

File: packages/mui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx [hunk 2]
```diff
@@ -38,54 +38,56 @@ export default function ArrayFieldItemTemplate<
...
-          {hasRemove && (
-            <RemoveButton
-              style={btnStyle}
-              disabled={disabled || readonly}
-              onClick={onDropIndexClick(index)}
-              uiSchema={uiSchema}
-              registry={registry}
-            />
-          )}
+    <Grid item={true} xs={12}>
```
Comment (heath-freenome) [inline packages/mui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx:41]: What about entirely removing the Grid `container` on [line 63](https://github.com/rjsf-team/react-jsonschema-form/blob/6ffc2ceb1f24e2661d0acbc0c3e09e5e443e9f27/packages/mui/src/ArrayFieldTemplate/ArrayFieldTemplate.tsx#L63) of `ArrayFieldTemplate` instead?

File: packages/material-ui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx (hunk 1 above)
Comment (jroebu14) [reply packages/material-ui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx:41]: Yes that makes more sense. There's just the question about what to do with the `key` prop on that Grid container. I could leave there and change the `Grid` to a `Box` or just remove it entirely along with the `key` prop. I'm not sure the `key` prop is necessary so maybe the latter makes more sense.

Comment (heath-freenome) [reply packages/material-ui/src/ArrayFieldItemTemplate/ArrayFieldItemTemplate.tsx:41]: Honestly the `key` prop is only required/used by React when it is being generated from some sort of map function, which this is not so you can remove it.

**Comment by heath-freenome** [comment]
@jroebu14 This looks great! Can you update the `CHANGELOG.md` to add sections for `@rjsf/material-ui` and `@rjsf/mui` underneath the `# 5.13.1` section

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,14 +15,22 @@ it according to semantic versioning. For example, if your PR adds a breaking cha
...
+## @rjsf/material-ui
+
+- Removed an unnecessary `Grid` container component in the `ArrayFieldTemplate` component that wrapped the `ArrayFieldItemTemplate`, fixing [#3863](https://github.com/rjsf-team/react-jsonschema-form/issues/3863)
+
 ## @rjsf/utils
 
 - Added `getOptionMatchingSimpleDiscriminator()` function
 - `getMatchingOption` and `getClosestMatchingOption` now bypass `validator.isValid()` calls when simple discriminator is provided, fixing [#3692](https://github.com/rjsf-team/react-jsonschema-form/issues/3692)
 
-
```
Comment (heath-freenome) [inline CHANGELOG.md:25]: All changes after this point should be reverted... I'm guessing your editor changed the formatting

Comment (jroebu14) [reply CHANGELOG.md:25]: ah oops! will fix just now 😄

**Review by heath-freenome — APPROVED** [review]

**Pull request merged**